lib/view_model/access_control.rb

```diff
@@ -89,17 +88,22 @@ def visible!(view, context:)
     end
   end
 
-  # Implementations of deserialization must call this when they know what
-  # changes are to be made to the viewmodel. For viewmodels with transactional
-  # backing models, the changes may be made in advance to give the edit checks
-  # the opportunity to compare values. Must be called with the saved
-  # `initial_editability` value if changes have been made.
-  def editable!(view, initial_editability: nil, deserialize_context:, changes:)
-    return if ineligible(view)
+  before_deserialize do |view, deserialize_context:|
+    next if ineligible(view)
 
-    initial_editability ||= editable_check(view, deserialize_context: deserialize_context)
+    initial_result =
+      editable_check(view, deserialize_context: deserialize_context)
 
-    result = initial_editability.merge do
+    save_editability(view, initial_result)
+  end
+
+  on_change do |view, changes:, deserialize_context:|
+    next if ineligible(view)
+    initial_result = fetch_editability(view)
+
+    next if changes.nil?
```

Comment:
Will this case ever be hit?

Reply:
not any more, after introducing the onChange hook. deleted.